airflow/contrib/hooks/aws_glue_catalog_hook.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+from airflow.contrib.hooks.aws_hook import AwsHook
+
+
+class AwsGlueCatalogHook(AwsHook):
+    """
+    Interact with AWS Glue Catalog
+
+    :param aws_conn_id: ID of the Airflow connection where
+        credentials and extra configuration are stored
+    :type aws_conn_id: str
+    :param region_name: aws region name (example: us-east-1)
+    :type region_name: str
+    """
+
+    def __init__(self,
+                 aws_conn_id='aws_default',
+                 region_name=None,
+                 *args,
+                 **kwargs):
+        self.region_name = region_name
```

Comment:
Remove region_name please - the AwsHook already supports pulling this from the extra field of the connection

Reply:
Yes, but sometimes you may want to use a region other than the default for the connection.

Reply:
What @ashb is saying is that you already can pass `region_name` via the `extra` field of the connection.
See this: https://github.com/apache/incubator-airflow/blob/master/airflow/contrib/hooks/aws_hook.py#L119
So if you want to connect to other regions just add a different connection.

Reply:
Okay, I see. You can pass it via the client_type to also be able to use different regions for one connection. I mean when there is the possibility via the `get_client_type` and you do not remove the possibility to set it via the connection. I am fine with it. Looks good to me, actually :D